From: Sage Weil <sage@newdream.net>
Date: Tue, 9 Aug 2011 15:05:07 -0700
Subject: [PATCH 14/124] libceph: warn on msg allocation failures

Git-commit: f0ed1b7cef1e801ef470efc501f9c663fe10cebd
Patch-mainline: v3.2
References: FATE#312404

Any non-masked msg allocation failure should generate a warning and stack
trace to the console.  All of these need to eventually be replaced by
safe preallocation or msgpools.

Signed-off-by: Sage Weil <sage@newdream.net>
Signed-off-by: Guangliang Zhao <gzhao@suse.com>
---
 net/ceph/messenger.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/net/ceph/messenger.c b/net/ceph/messenger.c
index 2de711a..f56aca3 100644
--- a/net/ceph/messenger.c
+++ b/net/ceph/messenger.c
@@ -2352,6 +2352,7 @@ out:
 	if (!can_fail) {
 		pr_err("msg_new can't create type %d front %d\n", type,
 		       front_len);
+		WARN_ON(1);
 	} else {
 		dout("msg_new can't create type %d front %d\n", type,
 		     front_len);
-- 
1.7.3.4

